src/core/libmaven/mzUtils.cpp

```diff
@@ -7,274 +7,210 @@
-                                                                                                 ...
-                                                                                                 In this example, the filter x is symmetric, with index of first sample = -2.
-
-                                                                                                 This function is optimized for architectures that can simultaneously perform
-                                                                                                 a multiply, add, and one load from memory; e.g., the IBM RISC System/6000.
-                                                                                                 Because, for each value of i, it accumulates the convolution sum z[i] in a
-                                                                                                 scalar, this function is not likely to be optimal for vector architectures.
-                                                                                                 ******************************************************************************
-                                                                                                 Author:  Dave Hale, Colorado School of Mines, 11/23/91
-                                                                                                 *****************************************************************************/
-    {
-        int ilx=ifx+lx-1,ily=ify+ly-1,ilz=ifz+lz-1,i,j,jlow,jhigh;
+    void conv (int xLen, int indexFirstX, float *x, int inputLen,
+              int indexFirstInput, float *input, int resultLen,
+              int indexFirstResult, float *result)
+   {
+        int ilx = indexFirstX + xLen - 1;
+        int ily = indexFirstInput + inputLen - 1;
+        int ilz = indexFirstResult + resultLen - 1;
+
+        int i, j, jlow, jhigh;
         float sum;
 
-        x -= ifx;  y -= ify;  z -= ifz;
-        for (i=ifz; i<=ilz; ++i) {
-            jlow = i-ily;  if (jlow<ifx) jlow = ifx;
-            jhigh = i-ify;  if (jhigh>ilx) jhigh = ilx;
-            for (j=jlow,sum=0.0; j<=jhigh; ++j)
-                sum += x[j]*y[i-j];
-            z[i] = sum;
+        x -= indexFirstX;
+        input -= indexFirstInput;
+        result -= indexFirstResult;
+
+        for (i = indexFirstResult; i <= ilz; ++i)
+        {
+            jlow = i - ily;
+            if (jlow < indexFirstX)
+                    jlow = indexFirstX;
+            jhigh = i - indexFirstInput;
+            if (jhigh > ilx)
+                jhigh = ilx;
+
+            for (j = jlow, sum = 0.0; j <= jhigh; ++j)
+                sum += x[j] * input[i-j];
+            result[i] = sum;
         }
     }
 
-    void gaussian1d_smoothing (int ns, int nsr, float *data)
+    void gaussian1d_smoothing (int numSample, int smoothWindowLen, float *data)
```

Comment:
Again, if `data` is the value of interest, then it should be returned and not passed as a pointer reference.

Reply:
This wasn't done because data array contains the values which are used and being edited within the function.

Reply:
No problem.